### PR TITLE
OOP School Library: Add Person, Student and Teacher Classes

### PR DESCRIPTION
**Pull request by peterdtitan**
## Created a Person class and did the following:

- [ ] Added instance vars `@id`, `@name`, parent_permission and `@age `.
- [ ] name and parent_permission are optional
- [ ] Create `@id`, `@name` and `@age` getters and ` @name` and `@age` setters.
- [ ] Added private and public methods `of_age? ` and `can_use_services? `

 ## Created Student class that inherits Person above.

- [ ] Extended the parent constructor by adding `@classroom`
- [ ] Created a `play_hooky` method

 ## Created a class Teacher that inherits Person class also.
- Extended parent constructor by adding `@specialization` and overriding the `can_use_services` method to always return true.

Screenshot 2023-05-25 at 12 57 08 PM

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Person
+  def initialize(age, name = 'unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+  attr_accessor :name, :age, :parent_permission
```
Comment (franclobo) [inline person.rb]: - [ ] Kindly move these lines before the initialize method, you can see this article of [how to use...](https://www.rubyguides.com/2018/11/attr_accessor/)

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Student < Person
+  attr_reader :classroom, :parent_permission
```
Comment (franclobo) [inline student.rb]: - [ ] Please 🙏🏽 remove this line, it is not necessary to use attr_reader for any variable in this class, you can see this article of [how to use...](https://www.rubyguides.com/2018/11/attr_accessor/)

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @peterdtitan, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (peterdtitan) [reply person.rb]: Done, thanks.

File: student.rb (hunk 2 above)
Comment (peterdtitan) [reply student.rb]: Done, thanks for the feedback.

**Review by franclobo — APPROVED** [review]
Hi @peterdtitan,

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Well done using atrr_reader, attr_writer, attr_accessor 💯 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**